awscli/customizations/s3/utils.py

```diff
@@ -760,6 +760,33 @@ def on_queued(self, future, **kwargs):
         future.meta.provide_transfer_size(self.size)
 
 
+class OnDoneFilteredSubscriber(BaseSubscriber):
```

Comment:
Yeah actually I think I want to add it to ``s3transfer``. When I was writing these subscribers, I just found it incredibly annoying that I had to keep adding this logic so I can see it being useful for others. I did not add it to ``s3transfer`` because if I add it there, then it would be a public interface for ``s3transfer`` with the CLI at least I can hide it from other developers from using, but yet again ``s3transfer`` is a work in progress. I just want to put a little more design thought into it. It was also a little difficult getting cross dependency of a PR in ``s3transfer`` with this ongoing PR.

What do you think? I am thinking either adding a TODO to make sure I backport this to ``s3transfer``, or do the backport now, wait till s3transfer pr gets merged, then update this PR. I think I am fine with either as both have their pros and cons.

I could also make the methods private if it is just the public part that you find strange.